packages/dev/core/src/Rendering/objectRenderer.ts

```diff
@@ -0,0 +1,684 @@
+    };
+
+    /**
+     * Define the list of particle systems to render. If not provided, will render all the particle systems of the scene.
+     * Note that the particle systems are rendered only if renderParticles is set to true.
+     */
+    public particleSystemList: Nullable<Array<IParticleSystem>> = null;
+
+    /**
+     * Use this function to overload the renderList array at rendering time.
+     * Return null to render with the current renderList, else return the list of meshes to use for rendering.
+     * For 2DArray, layerOrFace is the index of the layer that is going to be rendered, else it is the faceIndex of
+     * the cube (if the RTT is a cube, else layerOrFace=0).
+     * The renderList passed to the function is the current render list (the one that will be used if the function returns null).
+     * The length of this list is passed through renderListLength: don't use renderList.length directly because the array can
+     * hold dummy elements!
+     */
+    public getCustomRenderList: (layerOrFace: number, renderList: Nullable<Immutable<Array<AbstractMesh>>>, renderListLength: number) => Nullable<Array<AbstractMesh>>;
+
+    /**
+     * Define if particles should be rendered.
+     */
+    public renderParticles = true;
+
+    /**
+     * Define if sprites should be rendered.
+     */
+    public renderSprites = false;
+
+    /**
+     * Force checking the layerMask property even if a custom list of meshes is provided (ie. if renderList is not undefined)
+     */
+    public forceLayerMaskCheck = false;
+
+    /**
+     * Define the camera used to render the objects.
+     */
+    public activeCamera: Nullable<Camera>;
+
+    /**
+     * Override the mesh isReady function with your own one.
+     */
+    public customIsReadyFunction: (mesh: AbstractMesh, refreshRate: number, preWarm?: boolean) => boolean;
+
+    /**
+     * Override the render function with your own one.
+     */
+    public customRenderFunction: (
+        opaqueSubMeshes: SmartArray<SubMesh>,
+        alphaTestSubMeshes: SmartArray<SubMesh>,
+        transparentSubMeshes: SmartArray<SubMesh>,
+        depthOnlySubMeshes: SmartArray<SubMesh>,
+        beforeTransparents?: () => void
+    ) => void;
+
+    /**
+     * An event triggered before rendering the objects
+     */
+    public onBeforeRenderObservable = new Observable<number>();
+
+    /**
+     * An event triggered after rendering the objects
+     */
+    public onAfterRenderObservable = new Observable<number>();
+
+    /**
+     * An event triggered before the rendering group is processed
+     */
+    public onBeforeRenderingManagerRenderObservable = new Observable<number>();
+
+    /**
+     * An event triggered after the rendering group is processed
+     */
+    public onAfterRenderingManagerRenderObservable = new Observable<number>();
+
+    /**
+     * An event triggered when fast path rendering is used
+     */
+    public onFastPathRenderObservable = new Observable<number>();

```

Comment:
```suggestion
    public readonly onBeforeRenderObservable = new Observable<number>();

    /**
     * An event triggered after rendering the objects
     */
    public readonly onAfterRenderObservable = new Observable<number>();

    /**
     * An event triggered before the rendering group is processed
     */
    public readonly onBeforeRenderingManagerRenderObservable = new Observable<number>();

    /**
     * An event triggered after the rendering group is processed
     */
    public readonly onAfterRenderingManagerRenderObservable = new Observable<number>();

    /**
     * An event triggered when fast path rendering is used
     */
    public readonly onFastPathRenderObservable = new Observable<number>();

```

Reply:
Seems like there are a number of other properties that are never reassigned after the constructor and could be readonly.

Reply:
Same as above, it would be a breaking change, the user can reassign a new value to the property (even if I don't see why they would do that...).

Reply:
We finally decided with @sebavan that it should not break any code, so I did the change.